### PR TITLE
add boringssl build to travis

### PR DESCRIPTION
**Pull request by cpapazian**
testing a travis build with boringssl. see: https://github.com/google/libsxg/issues/78

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -280,7 +281,7 @@ function (configure_test test_name)
 endfunction ()
 
 if (RUN_TEST)
-  configure_test (nfail_malloc_test)
+  #configure_test (nfail_malloc_test)
```
Comment (cpapazian) [inline CMakeLists.txt:284]: this test is failing to build with boringssl

Comment (twifkak) [reply CMakeLists.txt:284]: Seems like a net win, nonetheless. After merging, I'll file a bug to fix this (or worst-case, add a cmake option for it.)

**Review by twifkak — CHANGES_REQUESTED** [review]
Strange [Travis failure](https://travis-ci.org/github/google/libsxg/jobs/758573398):

```
/home/travis/build/google/libsxg/src/sxg_sig.c:167:60: error: expected ')'
      snprintf(integer_buffer, sizeof(integer_buffer), "%" PRIu64, num);
                                                           ^
```

Any idea why? (Is this just downstream confusion because of the earlier "implicit function declaration" warnings?)

**Comment by cpapazian** [comment]
> Strange [Travis failure](https://travis-ci.org/github/google/libsxg/jobs/758573398):
> 
> ```
> /home/travis/build/google/libsxg/src/sxg_sig.c:167:60: error: expected ')'
>       snprintf(integer_buffer, sizeof(integer_buffer), "%" PRIu64, num);
>                                                            ^
> ```
> 
> Any idea why? (Is this just downstream confusion because of the earlier "implicit function declaration" warnings?)

note sure. i'm not getting it locally.

**Comment by twifkak** [comment]
You might try adding `#include <inttypes.h>` to `sxg_sig.c`. My guess is that:
1. The compiler is seeing an undefined `PRIu64` and assumes it's a variable reference, therefore parse error.
2. OpenSSL transitively includes inttypes.h and BoringSSL doesn't.
3. gcc magically includes inttypes.h into all translation units and clang doesn't.

File: .travis.yml [hunk 2]
```diff
@@ -2,16 +2,29 @@ language: cpp
...
+        - cd build
+        - cmake -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_C_COMPILER=$CC ..
+        - make VERBOSE=1
+        - ctest
+    - name: 'boringssl'
+      os: linux
+      compiler: clang
+      before_script:
+        - source ./.travis/install_cmake.sh
+        - source ./.travis/install_go.sh
```
Comment (cpapazian) [inline .travis.yml:22]: @twifkak let me know if there's a preferred way to install the golang dependency

Comment (twifkak) [reply .travis.yml:22]: Nope, that works for me. I guess your previous commit using the Travis `go: 1.13` directive didn't work?

**Comment by twifkak** [comment]
Thanks for this! I'll merge now. I filed #80 for nfail_malloc_test.

**Review by twifkak — APPROVED** [review]

File: .travis.yml (hunk 2 above)
Comment (cpapazian) [reply .travis.yml:22]: ya. i think the `go:` directive is just for go environments. docs were sparse, though ...

**Pull request merged**